pyuvdata/uvdata.py

```diff
@@ -1433,3 +1433,212 @@ def reorder_pols(self, order=None, run_check=True, run_check_acceptability=True)
+            blt_ind2 = self.antpair2ind(key[1], key[0])
+            if len(blt_ind1) + len(blt_ind2) == 0:
+                raise KeyError('Antenna pair {pair} not found in '
+                               'data'.format(pair=(key[0], key[1])))
+            if type(key[2]) is str:
+                # pol is str
+                pol_ind = np.where(self.polarization_array == uvutils.polstr2num(key[2]))[0]
+            else:
+                # polarization number a la AIPS memo
+                pol_ind = np.where(self.polarization_array == key[2])[0]
+            if len(pol_ind) == 0:
+                raise KeyError('Polarization {pol} not found in data.'.format(pol=key[2]))
+        return (blt_ind1, blt_ind2, pol_ind)
+
+    def get_data(self, key, squeeze=True):
+        """
+        Function for quick access to numpy array with data corresponding to
+        a baseline and/or polarization.
+
+        Args:
+            key: Identifier of data. See _key2inds for formatting.
+            squeeze: If true (default) remove single dimensional entries from output array.
+
+        Returns:
+            Numpy array of data corresponding to key.
+            If data exists conjugate to requested antenna pair, it will be conjugated
+            before returning.
+        """
+        ind1, ind2, indp = self._key2inds(key)
+        out = self.data_array[:, :, :, indp]
+        out = np.append(out[ind1, :, :, :], np.conj(out[ind2, :, :, :]), axis=0)
+        if squeeze:
+            out = np.squeeze(out)
+        return out
+
+    def get_flags(self, key, squeeze=True):
+        """
+        Function for quick access to numpy array with flags corresponding to
+        a baseline and/or polarization.
+
+        Args:
+            key: Identifier of data. See _key2inds for formatting.
+            squeeze: If true (default) remove single dimensional entries from output array.
+
+        Returns:
+            Numpy array of flags corresponding to key.
+        """
+        ind1, ind2, indp = self._key2inds(key)
+        out = self.flag_array[:, :, :, indp]
+        out = np.append(out[ind1, :, :, :], np.conj(out[ind2, :, :, :]), axis=0)
+        if squeeze:
+            out = np.squeeze(out)
+        return out
+
+    def get_nsamples(self, key, squeeze=True):
+        """
+        Function for quick access to numpy array with nsamples corresponding to
+        a baseline and/or polarization.
+
+        Args:
+            key: Identifier of data. See _key2inds for formatting.
+            squeeze: If true (default) remove single dimensional entries from output array.
+
+        Returns:
+            Numpy array of nsamples corresponding to key.
+        """
+        ind1, ind2, indp = self._key2inds(key)
+        out = self.nsample_array[:, :, :, indp]
+        out = np.append(out[ind1, :, :, :], np.conj(out[ind2, :, :, :]), axis=0)
+        if squeeze:
+            out = np.squeeze(out)
+        return out
+
+    def __getitem__(self, key):
```

Comment:
FWIW, I like the __getitem__ syntax a lot. From my standpoint, getting out a particular piece of data to then do some calculation on it is the first thing I want to do with a pyuvdata object and facilitating that in as simple and obvious a way as possible will go a long way toward getting people up and running quickly on pyuvdata. This is the design of DataContainer and part of what makes it such a breeze to use. 

I like that we have both .get_data() and .get_flags() in addition to __getitem__(). I support Adam's change.